COMPOUND_COMPONENTS.md

```diff
@@ -0,0 +1,336 @@
+# Compound Components
+
+## What is a compound component?
+
+It is a way of describing a complex component by simultaneously describing sub-components. A compound component is made up of a few different parts.
+
+* Container component
+* Sub-components
+* Shared model (optional)
+* Behavior hooks (optional)
+
+A compound component contrasts with a configuration component.
+
+Configuration component:
+```tsx
+<Tabs items={[
+  { title: 'First Tab', content: 'First Tab Contents' },
+  { title: 'Second Tab', content: 'Second Tab Contents' },
+]} />
+```
+
+Compound component:
+```tsx
+<Tabs>
+  <Tabs.List>
+    <Tabs.Item>First Tab</Tabs.Item>
+    <Tabs.Item>Second Tab</Tabs.Item>
+  </Tabs.List>
+  <Tabs.Panel>First Tab Contents</Tabs.Panel>
+  <Tabs.Panel>Second Tab Contents</Tabs.Panel>
+</Tabs>
+```
+
+In this example, `Tabs` is the container component and `Tabs.List` is a sub-component.
+
+Compound components always have a container component and sub-components. Components that do not contain sub-components are not considered compound components. Some compound components might not contain state or behavior. An example might be an `IconButton` which is a button that contains an icon. It might be a compound component only for styling purposes, but doesn't contain any special state or behaviors:
```

Comment:
I don't like the way this reads:

> Compound components always have a container component and sub-components. Components that do not contain sub-components are not considered compound components.

Those 2 sentences make the same statement. This is also covered by a statement earlier:

> [Compound components are] a way of describing a complex component by simultaneously describing sub-components.

Those 2 sentences do not directly relate to the rest of the paragraph, which talks about not having a model or behavior. Perhaps remove those sentences or move them somewhere else.

```suggestion
Some compound components might not contain state or behavior. An example might be an `IconButton` which is a button that contains an icon. It might be a compound component only for styling purposes, but doesn't contain any special state or behaviors:
```